scripts/sct_testing.py

```diff
@@ -134,45 +179,29 @@ def main(args=None):
 
     # get list of all scripts to test
     list_functions = fill_functions()
-    if function_to_test:
-        if function_to_test in list_functions:
-            # overwrite variable to include only the function to test
-            list_functions = [function_to_test]
+    if functions_to_test:
+        for f in functions_to_test:
+            if f not in list_functions:
+                sct.printv('Command-line usage error: Function "%s" is not part of the list of testing functions' % function_to_test, type='error')
+        list_functions = functions_to_test
+
+    if jobs != 1:
+        pool = multiprocessing.Pool(processes=jobs)
+
+        results = list()
+        # loop across functions and run tests
+        for f in list_functions:
+            res = pool.apply_async(process_function, (f, param,))
+            results.append(res)
+
+    for idx_function, f in enumerate(list_functions):
```

Comment:
My only concern here is if a message is printed from inside the testing, that will appear here out-of-order, causing confusion on which testing the error message came from because it is executing all of them in parallel.
One solution to that would be capturing and returning the `stdout` and `stderrr` inside the `process_function()`, so you can print these messages here later in order.

Reply:
Not sure what you mean, the messages are indeed collected and (in case of error) printed in order.

Reply:
As far as I understood, it is capturing only exceptions but not any other message such as prints, logging, etc. That's my concern.